### PR TITLE
Game start endpoint PR

### PR DESCRIPTION
**Pull request by sremedios**
# Summary of changes
- Overall refactored router
- Added exe_args: Vec<String> to the config so that we can start games with special command-line arguments
- Disabled serde serialization for exe_path and exe_args
- Endpoint at POST /api/v1/start_game
- Take in a parameter id that specifies the game ID
- Start the game referred to by the given ID using exe_path, exe_args, and std::process::Command

# How was this tested?
- Created test executables for two test games

# TODO
- fixing PR comments

### CODE REVIEW

File: arclightning_backend/src/main.rs [hunk 1]
```diff
@@ -26,46 +26,147 @@ struct Game {
...
+        let response = req_body
+            .concat2()
+            .map_err(|_e| {
+                io::Error::new(
+                    ErrorKind::Other,
+                    "Failed to acquire mutex lock on games list".to_owned(),
+                )
+            })
+            .map(|body| String::from_utf8(body.to_vec()).unwrap())
+            .and_then(move |game_id| {
```
[the remark this answers is not in the document]
Comment (sremedios) [reply arclightning_backend/src/main.rs]: same

File: arclightning_backend/src/router.rs [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+impl hyper::service::NewService for Router {
+    type ReqBody = Body;
+    type ResBody = Body;
+    type Error = io::Error;
+    type Service = Router;
+    type Future = Box<Future<Item = Self::Service, Error = Self::InitError> + Send>;
+    type InitError = Error;
+    fn new_service(&self) -> Self::Future {
+        Box::new(future::ok(Self {
+            // TODO: is cloning necessary here?
```
Comment (starfys) [inline arclightning_backend/src/router.rs]: yes

File: arclightning_backend/src/router.rs [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+            games_list: Arc::new(Mutex::new(games_list)),
+        }
+    }
+
+    fn invalid_endpoint(&self) -> ResponseFuture {
+        Box::new(future::result(
+            Response::builder()
+                .status(StatusCode::NOT_FOUND)
+                .body(Body::from("uwu 404 not foundu hiss".to_owned()))
+                .map_err(|_e| {
```
Comment (starfys) [inline arclightning_backend/src/router.rs]: You can just use `_` here

File: arclightning_backend/src/router.rs [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+                    io::Error::new(
+                        ErrorKind::Other,
+                        "Failed to acquire mutex lock on games list".to_owned(),
```
Comment (starfys) [inline arclightning_backend/src/router.rs]: *hiss* wrong error message

File: arclightning_backend/src/router.rs [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+    fn start_game(&self, req_body: Body) -> ResponseFuture {
+        let games_list = self.games_list.clone();
+
+        let response = req_body
+            .concat2()
+            .map_err(|err| {
+                io::Error::new(
+                    ErrorKind::Other,
+                    format!("Failed to acquire mutex lock on games list: {}", err).to_owned(),
```
Comment (starfys) [inline arclightning_backend/src/router.rs]: incorrect error!

File: arclightning_backend/src/game.rs [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    pub exe_args: Vec<String>,
+}
+
+pub fn toml_to_hashmap(toml_filepath: &PathBuf) -> Result<HashMap<String, Game>, io::Error> {
+    let mut games_toml = String::new();
+    let mut file = File::open(&toml_filepath)?;
+    file.read_to_string(&mut games_toml)?;
+
+    // error casting for homogeneous errors
+    toml::from_str(&games_toml).map_err(|e| io::Error::new(ErrorKind::Other, e))
```
Comment (starfys) [inline arclightning_backend/src/game.rs]: change `e` to `err`

**Comment by starfys** [comment]
fix merge conflicts and we good

**Pull request merged**